haystack/components/websearch/searchapi.py

```diff
@@ -64,17 +79,25 @@ def to_dict(self) -> Dict[str, Any]:
     @classmethod
     def from_dict(cls, data: Dict[str, Any]) -> "SearchApiWebSearch":
         """
-        Deserialize this component from a dictionary.
+        Deserializes the component from a dictionary.
+
+        :param data:
+            The dictionary to deserialize from.
+        :returns:
+                The deserialized component.
         """
         deserialize_secrets_inplace(data["init_parameters"], keys=["api_key"])
         return default_from_dict(cls, data)
 
-    @component.output_types(documents=List[Document], links=List[str])
-    def run(self, query: str):
+    @component.output_types(documents=List[Document], links=Union[List[Document], List[str]])
+    def run(self, query: str) -> Dict[str, Union[List[Document], List[str]]]:
```

Comment:
The signature in the `Component` protocol is this one.

```suggestion
    def run(self, query: str) -> Dict[str, Any]:
```

Reply:
And we can't be more precise? Still adheres to the protocol, no?

Reply:
Yeah, true. Still makes sense and shouldn't cause any issues.